pkg/resources/shared_database.go

```diff
@@ -0,0 +1,266 @@
+	},
+	"log_level": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		ValidateDiagFunc: StringInSlice(sdk.AsStringList(sdk.AllLogLevels), true),
+		DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+			return d.Get(k).(string) == "OFF" && newValue == ""
+		},
+		Description: fmt.Sprintf("Specifies the severity level of messages that should be ingested and made available in the active event table. Valid options are: %v. Messages at the specified level (and at more severe levels) are ingested. For more information, see [LOG_LEVEL](https://docs.snowflake.com/en/sql-reference/parameters.html#label-log-level).", sdk.AsStringList(sdk.AllLogLevels)),
+	},
+	"trace_level": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		ValidateDiagFunc: StringInSlice(sdk.AsStringList(sdk.AllTraceLevels), true),
+		DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+			return d.Get(k).(string) == "OFF" && newValue == ""
+		},
+		Description: fmt.Sprintf("Controls how trace events are ingested into the event table. Valid options are: %v. For information about levels, see [TRACE_LEVEL](https://docs.snowflake.com/en/sql-reference/parameters.html#label-trace-level).", sdk.AsStringList(sdk.AllTraceLevels)),
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the database.",
+	},
+}
+
+func SharedDatabase() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateSharedDatabase,
+		UpdateContext: UpdateSharedDatabase,
+		ReadContext:   ReadSharedDatabase,
+		DeleteContext: DeleteSharedDatabase,
+		Description:   "A shared database creates a database from a share provided by another Snowflake account. For more information about shares, see [Introduction to Secure Data Sharing](https://docs.snowflake.com/en/user-guide/data-sharing-intro).",
+
+		Schema: sharedDatabaseSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func CreateSharedDatabase(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id := sdk.NewAccountObjectIdentifier(d.Get("name").(string))
+	externalShareId := sdk.NewExternalObjectIdentifierFromFullyQualifiedName(d.Get("from_share").(string))
+
+	var externalVolume *sdk.AccountObjectIdentifier
+	if v, ok := d.GetOk("external_volume"); ok {
+		externalVolume = sdk.Pointer(sdk.NewAccountObjectIdentifier(v.(string)))
+	}
+
+	var catalog *sdk.AccountObjectIdentifier
+	if v, ok := d.GetOk("catalog"); ok {
+		catalog = sdk.Pointer(sdk.NewAccountObjectIdentifier(v.(string)))
+	}
+
+	var logLevel *sdk.LogLevel
+	if v, ok := d.GetOk("log_level"); ok {
+		logLevel = sdk.Pointer(sdk.LogLevel(v.(string)))
```

Comment:
Nit: is it okay with casting like that? Users can provide invalid value, we could validate this with `sdk.LogLevelFromString` or something

Reply:
We are validating input on the schema level, so only valid values should be passed here